host/CMakeLists.txt

```diff
@@ -488,3 +488,321 @@ install(TARGETS oehostapp EXPORT openenclave-targets)
+		  #
+		  #      sgx/windows/aep.asm
+		  #      ${CMAKE_CURRENT_BINARY_DIR}/enter.obj
+		  #      sgx/windows/exception.c
+		  #      sgx/windows/switchless.c
+      sgx/windows/xstate.c)
+  endif()
+
+  set(PLATFORM_FLAGS "-m64")
+elseif(OE_TRUSTZONE)
+	list(APPEND PLATFORM_OEMR_SRC
+    optee/log.c)
+
+  if (UNIX)
+	  list(APPEND PLATFORM_OEMR_SRC
+      optee/linux/enclave.c)
+  else()
+    message(FATAL_ERROR "OP-TEE is not yet supported on platforms other than Linux.")
+  endif()
+
+  set(PLATFORM_FLAGS "")
+endif()
+
+if (OE_SGX AND WIN32)
+  # oedebugrt is accessed via a bridge on Win32 and need not be linked.
+  list(APPEND PLATFORM_OEMR_SRC
+    sgx/windows/debugrtbridge.c)
+
+  # __oe_dispatch_ocall function must have debug symbols for debuggers to
+  # stitch the ocall stack. Use /Z7 flag to embed the debugging information
+  # in the object file. When a host application is built, the linker will
+  # extract this debugging information and put it in the pdb for the host
+  # application. Using /Z7 avoids having to distribut oehost.pdb.
+  # Note: This is a temporary fix and debuggers will not have this requirement
+  # when ocall stack stitching is also done via the debugger contract.
+  set_source_files_properties(
+    sgx/calls.c
+    PROPERTIES COMPILE_FLAGS "/Z7")
+endif()
+
+# Common host verification files that work on any OS/architecture.
+list(APPEND PLATFORM_OEMR_SRC
+	#  ../common/attest_plugin.c
+	# ../common/datetime.c
+	  ../common/safecrt.c
+  hexdump.c
+  dupenv.c
+  fopen.c
+  #  tests.c
+  result.c
+  traceh.c)
+
+# Common files that are used in the OE SDK only.
+list(APPEND PLATFORM_OEMR_SRC
+	#  ../common/kdf.c
+	#  ../common/argv.c
+	#  asym_keys.c
+#  calls.c
+#  ocalls.c
+#  error.c
+#  files.c
+#  fopen.c
+  memalign.c
+#  syscall_u_wrapper.c
+#  signkey.c
+  strings.c
+  #  tee_u_wrapper.c
+  #traceh_enclave.c)
+  )
+
+add_custom_command(
+    OUTPUT ${CMAKE_SOURCE_DIR}/sgx/create.c
+    DEPENDS sgx_u.h
+    COMMAND "")
+
+add_library(oemr STATIC
+	#  ${PLATFORM_HOST_ONLY_SRC}
+	#  ${PLATFORM_SDK_ONLY_SRC})
+	${PLATFORM_OEMR_SRC})
+
+target_link_libraries(oemr PUBLIC oe_includes)
+
+if(WIN32)
+  target_link_libraries(oemr PUBLIC ws2_32)
+  target_link_libraries(oemr PUBLIC Bcrypt)
+  target_link_libraries(oemr PUBLIC crypt32)
+endif()
+
+#if (OE_SGX AND UNIX)
+#  # Link oedebugrt static library.
+#  target_link_libraries(oemr PRIVATE oedebugrt)
+#
+#  # enter.c must be forced to retain the frame-pointer
+#  # for ocall stack-stitching by using the -fno-omit-frame-pointer flag.
+#  # It is compiled with -O2 flag to retain the same generated assembly
+#  # code in both debug and release builds.
+#  set_source_files_properties(oemr
+#    sgx/enter.c
+#    PROPERTIES
+#    COMPILE_FLAGS "-O2 -fno-omit-frame-pointer")
+#endif()
+
+#add_dependencies(oemr syscall_untrusted_edl)
+#add_dependencies(oemr tee_untrusted_edl)
+if(OE_SGX)
+  add_dependencies(oemr sgx_untrusted_edl switchless_untrusted_edl)
```

Comment:
oesign most likely does not need any edl stuff either.

Reply:
Since sgx_untrusted_edl  depends on sgx_u.h, add_dependencies can ensure dependency on sgx_u.h. I will try to find if there is a straightforward way for dependency on file.